modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/AbstractCSharpCodegen.java

```diff
@@ -307,10 +307,26 @@ public void processOpts() {
                 }
             }
         }
+
+        processUsings(objs);
+
         // process enum in models
         return postProcessModelsEnum(objs);
     }
 
+    private void processUsings(Map<String, Object> objs) {
```

Comment:
Hi. I've had a bit of rethink. I 'think' I have a solution, i.e. it works for me :),  but I'm going to pull this PR for now and resubmit after a bit more testing.